docs/hub-deployment-guide/configure-auth/github-orgs.md

```diff
@@ -230,3 +230,33 @@ To enable this access,
     that profile. Add `2i2c-org:hub-access-for-2i2c-staff` to all
     `allowed_teams` so 2i2c engineers can log in to debug issues. If
     `allowed_teams` is not set, that profile is not available to anyone.
+
+### Enabling team based access on hub with pre-existing users
+
+If this is being enabled for users on a hub with *pre-existing* users, they
+will all need to be logged out before deployment. This would force them to
+re-login next time, and that will set `auth_state` properly so we can filter
+based on team membership - without that, we won't know which teams the user
+belongs to, and they will get an opaque 'Access denied' error.
+
+1. Check with the community to know *when* is a good time to log everyone
+   out. If users have running servers, they will need to refresh the page -
+   which will put them through the authentication flow again. It's best to
+   do this at a time when minimal or no users are running, to minimze
+   disruption.
+
+2. We log everyone out by regenerating [hub.cookieSecret](https://z2jh.jupyter.org/en/stable/resources/reference.html#hub-cookiesecret).
```

Comment:
Presuming I have the use case correct (there is a hub that was already using GitHub teams/org auth so already have a GitHub OAuth App, and now they want teams-based access to profiles) and there isn't anything more complicated happening at the hub-level, I actually think there is an easier way to do this that does not involve using `kubectl` at all:

1. Go to the GitHub OAuth App settings page for the hub, e.g., Openscapes staging is here https://github.com/organizations/2i2c-org/settings/applications/2367936
2. Click "Revoke all user tokens"

This will also force the oauth flow and make everyone login again.

We have this documented [here](https://infrastructure.2i2c.org/sre-guide/common-problems-solutions/#resetting-a-github-oauth-app-when-users-are-faced-with-a-403-forbidden-error-at-login) as a method of debugging when someone did not correctly authorise the GitHub OAuth App on first login.

Reply:
@sgibson91 Thanks for pointing that out - I had not considered that! It would indeed be much simpler!  However, I just tried that and it did not work :( I think it works for the 'Authorize' use case because they are not JupyterHub users yet at that point. In this case, since they've already been successfully authenticated into *JupyterHub*, revoking GitHub creds does nothing, since they're using JupyterHub cookies / creds at that point. It would only take effect after they explicitly log out of *JupyterHub*. So we need to rotate cookie secret instead.